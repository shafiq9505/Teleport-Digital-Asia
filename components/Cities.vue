<template>
	<b-card no-body>
		<b-card-header>
			<strong>Select Cities</strong>
		</b-card-header>
		<b-card-body class="pl-2">
			<b-container>
				<!-- Cities Selection -->
				<b-row>
					<b-col v-show="fields.citiesSelection.fieldAttr.show">
						<b-form-group horizontal :label-cols="4" label="Please select a cities">
							<select class="form-control" v-model="formData.citiesModel" v-on:change="onChangeCities()">
								<option v-for="option in fields.citiesSelection.options" :key="option.city" :value="option.city">{{ option.city }}</option>
							</select>
						</b-form-group>
					</b-col>
				</b-row>

				<Weather v-if="getWeather" :weatherObj="getWeather" />
			</b-container>
		</b-card-body>
	</b-card>
</template>
<script>
import cities from '../assets/json/cities.json'
import { mapGetters } from 'vuex'
export default {
	name: 'Cities',
	data() {
		return {
			formData: {
				citiesModel: '',
				weather: {
					humidity: '123',
					sky: 'clear'
				}
			},
			fields: {
				citiesSelection: {
					options: [],
					fieldAttr: {
						show: true
					}
				},
				weatherData: {
					options: [],
					fieldAttr: {
						show: true
					}
				}
			}
		}
	},
	computed: {
		...mapGetters({
			getWeather: 'weather/weatherStore/getWeather'
		})
	},
	async mounted() {
		await this.init()
	},
	methods: {
		async init() {
			this.fields.citiesSelection.options = _.cloneDeep(cities)
			this.formData.citiesModel = cities[0].city

			await this.getCities()
		},
		async onChangeCities() {
			await this.getCities()
		},
		async getCities() {
			try {
				await this.$store.dispatch('weather/weatherStore/fetchWeather', this.formData.citiesModel)
			} catch (err) {
				console.error(err)
			}
		}
	}
}
</script>
<style>
.NuxtLogo {
	animation: 1s appear;
	margin: auto;
}

@keyframes appear {
	0% {
		opacity: 0;
	}
}
</style>
