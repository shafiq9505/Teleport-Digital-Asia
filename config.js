module.exports = {
	APP_VERSION: 'x.y.z-sss',
    WEATHER_API_KEY : '283d16e34cf45af9d977478080dae918',
	URL_PATHS: {
		WEATHER: {
			LOGOUT: 'api.openweathermap.org/data/2.5/weather'
		},
	
	},

}
