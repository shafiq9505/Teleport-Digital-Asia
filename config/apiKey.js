export const apiKey = '283d16e34cf45af9d977478080dae918'