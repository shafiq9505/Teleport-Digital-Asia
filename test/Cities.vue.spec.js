import Cities from '@/components/Cities.vue'
import { shallowMount,createLocalVue } from '@vue/test-utils'
import { stubs } from './config/bootStrapVue'
import _ from 'lodash'
import cities from '../assets/json/cities.json'
const localVue = createLocalVue()
localVue.use(_)
jest.mock('lodash')
let wrapper
beforeEach(() => {
	wrapper = shallowMount(Cities, {
        mocks : {
            
        },
		stubs
	})
})

describe('data', () => {
	it('default assigned data',async () => {
        
		expect(Cities.data()).toMatchSnapshot()
        
	})
})

describe('mounted', () => {
	it('mounted data',async  () => {
        wrapper.vm.onChangeCities = jest.fn()
		expect(wrapper.vm.fields.citiesSelection.options).toEqual(cities)
        
	})
})

describe('methods', () => {
	describe('onChangeCities', () => {
		it('change cities',async () => {
            wrapper.vm.$emit = jest.fn()
           await  wrapper.vm.onChangeCities()
           expect(wrapper.vm.$emit).toBeCalledWith("onChangeCities",{city : 'Kuala Lumpur'})
		})
	})
})
