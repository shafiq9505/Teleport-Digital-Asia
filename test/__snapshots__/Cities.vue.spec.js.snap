// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`data default assigned data 1`] = `
Object {
  "fields": Object {
    "citiesSelection": Object {
      "fieldAttr": Object {
        "show": true,
      },
      "options": Array [],
    },
  },
  "formData": Object {
    "citiesModel": "",
  },
}
`;
